src/argus/incident/views.py

```diff
@@ -250,6 +254,64 @@ def ticket_url(self, request, pk=None):
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
+@extend_schema_view(
+    update=extend_schema(
+        request=EmptySerializer,
+    ),
+)
+class TicketPluginViewSet(viewsets.ViewSet):
+    permission_classes = [IsAuthenticated]
+    serializer_class = IncidentTicketUrlSerializer
+    write_serializer_class = EmptySerializer
+    queryset = Incident.objects.all()
+
+    def update(self, request, incident_pk=None):
+        incident = get_object_or_404(self.queryset, pk=incident_pk)
+
+        if incident.ticket_url:
+            serializer = self.serializer_class(data={"ticket_url": incident.ticket_url})
+            if serializer.is_valid():
+                return Response(serializer.data, status=status.HTTP_200_OK)
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        plugin = getattr(settings, "TICKET_PLUGIN", None)
+
+        if not plugin:
+            return Response(
+                data="No path to ticket plugin can be found in the settings. Please update the setting 'TICKET_PLUGIN'.",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            ticket_class = import_class_from_dotted_path(plugin)
+        except:
+            return Response(
+                data="Ticket plugin could not be imported. Please check that the path in 'TICKET_PLUGIN' is correct.",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
This way of handling the error masks every detail, and amounts to "something went wrong", which isn't very useful.

1. Don't ever use this broad an exception handler. In this case, you should at least do `except Exception:`.
2. This should really *log* what's going on here. If there is a problem importing the configured plugin, a full traceback should be logged, so the admin can at least try to figure out what's wrong (it's no concern of the API client, though)

Reply:
What he's saying is basically:
```suggestion
        try:
            ticket_class = import_class_from_dotted_path(plugin)
        except Exception as e:
            return Response(
                data=f"Ticket plugin could not be imported: {e}",
                status=status.HTTP_400_BAD_REQUEST,
            )
```

Reply:
BUT, afaik `import_class_from_dotted_path`  should normally raise `ImportError` so you can catch that one separately since that one is to be *expected*. `except Exception` is for the *unexpected*.